Do not install berkeley-db docs by default

haampie opened the pull request:
It's a link dep of perl, which is a build dep of cmake, and it's just annoyingly big for containers and slow to install on lustre filesystems.

Install size goes down from 100MB -> 8MB.

Tested it real quick, works on 6.1.29, 6.2.32, 18.1.40. Versions 18.1.32 and 6.0.35 are not available for download anymore, and 5.3.28 does not compile with GCC 10 but the patch applies alright. It's out of the scope of this PR to fix these build / download struggles.

haampie committed:
Add an option to not install 100MB of docs

adamjstewart commented:
> Versions 18.1.32 and 6.0.35 are not available for download anymore, and 5.3.28 does not compile with GCC 10 but the patch applies alright. It's out of the scope of this PR to fix these build / download struggles.

Want to deprecate them?

chuckatkins commented:
Instead of modifying the docs that get installed, how about just dropping the docs from the install?
```
diff --git a/Makefile.in.orig b/Makefile.in
index c9f4c32..e166c88 100644
--- a/Makefile.in.orig
+++ b/Makefile.in
@@ -1215,9 +1215,9 @@ db_log_verify: db_log_verify@o@ $(COMMON_UTIL_OBJS) $(DEF_LIB)
 # Library and standard utilities install.
 ##################################################
 library_install: install_setup
-library_install: install_include install_lib install_utilities @ADDITIONAL_INSTALL@ install_docs
+library_install: install_include install_lib install_utilities @ADDITIONAL_INSTALL@
 
-uninstall: uninstall_include uninstall_lib uninstall_utilities @ADDITIONAL_UNINSTALL@ uninstall_docs
+uninstall: uninstall_include uninstall_lib uninstall_utilities @ADDITIONAL_UNINSTALL@
 
 install_setup:
        @test -d $(DESTDIR)$(prefix) || \
```

haampie committed:
Add deprecations of versions that are removed and add conflicts for 5.3

haampie commented:
@chuckatkins that's possible too, but that patch wouldn't apply to all versions in one go, so too much of a hassle for me.

@adamjstewart I've added deprecations to those versions that don't download anymore and added conflicts for the compilers I've tested.
